internal/handlers/sqlite/msg_findandmodify.go

```diff
@@ -130,7 +130,12 @@ func (h *Handler) findAndModifyDocument(ctx context.Context, params *common.Find
 	closer := iterator.NewMultiCloser(iterator.CloserFunc(cancel))
 	defer closer.Close()
 
-	queryRes, err := c.Query(ctx, nil)
+	var qp backends.QueryParams
+	if !h.DisableFilterPushdown {
+		qp.Filter = params.Query
+	}
+
+	queryRes, err := c.Query(ctx, &qp)
```

Comment:
I guessed either this or `update` pushdown is necessary to speed up. If I'm wrong no need for this change.

> Modifies single field. We do it from Go without using any MongoDB tricks

Reply:
Both changes are correct. Let's add it to other places too (count, delete, etc)

Reply:
:+1: 

The only one that doesn't have pushdown is `aggregate` for now. Do we want to add for `aggregate` as well?

Reply:
Not now, I think that would be more complicated

Reply:
Sounds good :+1: